client/client_inventory.go

```diff
@@ -35,29 +35,37 @@ func NewInventory() InventorySubmitter {
 
 // Submit reports status information to the backend
 func (i *InventoryClient) Submit(api ApiRequester, url string, data interface{}) error {
-	req, err := makeInventorySubmitRequest(url, data)
-	if err != nil {
-		return errors.Wrapf(err, "failed to prepare inventory submit request")
-	}
+	// PATCH used to be the only method available in Mender Product 2.5, so
+	// fall back to that if PUT fails.
+	methods := []string{http.MethodPut, http.MethodPatch}
+	for _, method := range methods {
+		req, err := makeInventorySubmitRequest(method, url, data)
+		if err != nil {
+			return errors.Wrapf(err, "failed to prepare inventory submit request")
+		}
 
-	r, err := api.Do(req)
-	if err != nil {
-		log.Error("Failed to submit inventory data: ", err)
-		return errors.Wrapf(err, "inventory submit failed")
-	}
+		r, err := api.Do(req)
+		if err != nil {
+			log.Error("Failed to submit inventory data: ", err)
+			return errors.Wrapf(err, "inventory submit failed")
+		}
 
-	defer r.Body.Close()
+		r.Body.Close()
```

Comment:
This you cannot do actually. `NewAPIError` unmarshals the error message from the response body.

Reply:
Oh, good catch! I did it because `defer` doesn't scope correctly in loops, but maybe I'll just move the whole thing inside a function then.